### DISCUSSION TITLE
Define a macro if it is not defined

### DISCUSSION THREAD

**Discussion by vhaispdeaded** [rpm-software-management/rpm#2342] [Q&A] [answered]
I need help understanding the syntax of defining a macro if it is not defined, as described at [Macro syntax](https://rpm-software-management.github.io/rpm/manual/macros.html)
```
%{!?with_python3: %global with_python3 1}
```

I am trying to build the following macro:
```
%global _TSCOVersion %{!?major:%global major 22}.%{!?minor:%global minor 4}.%{!?servicepack:%global servicepack 00}
```
I expect it to expand to: ```22.4.00```
but it expands to:  ```..```

Am I misunderstanding the syntax?

I am using RPM 4.14.3 on Red Hat Enterprise Linux 8.7.

Thank you,

**Comment by pmatilai** [comment, marked as answer]
`%global` itself expands to empty, not the defined value. Eg.

```
[pmatilai🎩︎localhost ~]$ rpm --eval "%global major 22"

[pmatilai🎩︎localhost ~]$ rpm --eval "%global major 22" --eval "%{major}"

22

```
